clang/lib/Sema/SemaExpr.cpp

```diff
@@ -18112,7 +18112,8 @@ void Sema::MarkFunctionReferenced(SourceLocation Loc, FunctionDecl *Func,
 
         if (FirstInstantiation || TSK != TSK_ImplicitInstantiation ||
             Func->isConstexpr()) {
-          if (isa<CXXRecordDecl>(Func->getDeclContext()) &&
+          if (!Func->isConstexpr() &&
```

Comment:
Nit: Slight simplification, you might as well move this branch after the `else if (Func->isConstexpr())` below instead of testing it's negation.